### ISSUE TITLE
[Bug] Unit test for overridden properties in Magento.Files.LineLength doesn't work as expected

### ISSUE THREAD

**Issue by lenaorobei** [magento/magento-coding-standard#36]
Rule from `ruleset.xml`
``` xml
    <rule ref="Magento.Files.LineLength">
        <properties>
            <property name="lineLimit" value="120"/>
            <property name="absoluteLineLimit" value="120"/>
        </properties>
        <type>warning</type>
        <severity>6</severity>
    </rule>
```

Properties from parent `Generic.Files.LineLength` sniff are processed instead of changed in `Magento.Files.LineLength` values.

### Steps to reproduce
1. Create test fixture with line length = 101.

### Expected result
1. Unit test does not detect warning.

### Actual result
```
[LINE 2] Expected 0 error(s) and 1 warning(s) in LineLengthUnitTest.inc but found 1 error(s) and 0 warning(s). The error(s) found were:
 -> Line exceeds maximum limit of 100 characters; contains 101 characters (Magento.Files.LineLength.MaxExceeded)
```

Discovered while implementing [Unit test for LineLengthSniff](https://github.com/magento/magento-coding-standard/issues/6).

**Comment by mzeis** [comment]
I did some debugging. 

When executing `bin/phpcs --standard=Magento`, the setting works as expected. 

* `PHP_CodeSniffer\Runner` initialises the ruleset and the processing of the Magento coding standard is [triggered](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/src/Ruleset.php#L200). 
* [PHP_CodeSniffer\Ruleset::processRuleset()](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/src/Ruleset.php#L318) reads the complete ruleset.xml including the overridden property settings and the line limit is applied correctly.

When executing `bin/phpunit`, the code is executed differently.

* The phpcs test suite bootstrap [defines](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/tests/bootstrap.php#L10) a constant `PHP_CODESNIFFER_IN_TESTS`. 
* [PHP_CodeSniffer\Tests\Standards\AllSniffs::suite()](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/tests/Standards/AllSniffs.php#L41) gathers the tests for all installed, non-ignored coding standards and runs the tests.
* When running the sniffs, the abstract sniff unit test class [triggers](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/tests/Standards/AbstractSniffUnitTest.php#L147-L148) the processing of the Magento coding standard.
* But this time, the logic is [different](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.3.2/src/Ruleset.php#L177-L190). When running the test suite, not the complete coding standard is interpreted but only the rules mentioned explicitly. Also, this logic is only executed for the first sniff of the coding standard.
* I assume that the difference in the last step is the reason for this bug.

Do we know if this worked before for other sniffs / in other repositories?

**Comment by lenaorobei** [comment]
@mzeis thanks for the investigation. 
For me this CodeSniffer behaviour looks like expected. Since we override `Generic.Files.LineLengthSniff` we need to take care about its public properties. Unit tests cover not ruleset, but sniffs, so i believe the fix is to move property declaration from ruleset to the sniff itself.
